builtin/sampled_dataset.cc

```diff
@@ -0,0 +1,368 @@
+/** sampled_dataset.cc                                              -*- C++ -*-
+    Francois Maillet, 11 janvier 2016
+    This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
+
+*/
+
+#include "mldb/sql/sql_expression.h"
+#include "sampled_dataset.h"
+#include "mldb/jml/utils/lightweight_hash.h"
+#include "mldb/types/any_impl.h"
+#include "mldb/types/structure_description.h"
+#include <random>
+#include <unordered_set>
+
+using namespace std;
+
+
+namespace Datacratic {
+namespace MLDB {
+
+    
+/*****************************************************************************/
+/* SAMPLED DATASET CONFIG                                                    */
+/*****************************************************************************/
+
+std::random_device rd;
+
+SampledDatasetConfig::
+SampledDatasetConfig() :
+        rows(0), fraction(0), withReplacement(false)
+{
+    seed = rd();
+}
+
+
+/*****************************************************************************/
+/* SAMPLED DATASET CONFIG                                                 */
+/*****************************************************************************/
+
+DEFINE_STRUCTURE_DESCRIPTION(SampledDatasetConfig);
+
+SampledDatasetConfigDescription::
+SampledDatasetConfigDescription()
+{
+    nullAccepted = true;
+
+    addField("rows", &SampledDatasetConfig::rows,
+            "Number of rows to sample. It cannot be set when fraction is "
+            "set.");
+    addField("fraction", &SampledDatasetConfig::fraction,
+            "Fraction of rows in dataset to sample. It cannot be set when "
+            "rows is set.");
+    addField("withReplacement", &SampledDatasetConfig::withReplacement,
+            "Sample with or without replacement", false);
+    addField("dataset", &SampledDatasetConfig::dataset,
+             "Dataset to sample");
+    addField("seed", &SampledDatasetConfig::seed,
```

Comment:
We should add: "The purpose of this parameter is to permit reproducible random samples. This parameter is optional, with the default value being selected randomly for each sample. "